modules/apps/27-interchain-accounts/controller/ibc_middleware.go

```diff
@@ -26,27 +26,37 @@ var (
 // ICA controller keeper and the underlying application.
 type IBCMiddleware struct {
 	app    porttypes.IBCModule
-	keeper keeper.Keeper
+	keeper *keeper.Keeper
```

Comment:
I made the keepers inside middlewares and modules to now be pointers so that I can call `WithICS4Wrapper` on them and have it persist in other references

Reply:
If you are using a keeper function to do SendPacket and WriteAck then yes. We have middlewares that don't use keepers (e.g. callbacks) so then its not needed

Reply:
OK, sounds like it will be quite important to get some migration docs for all of this then!